cli/src/semgrep/formatter/osemgrep_sarif.py

```diff
@@ -78,6 +78,8 @@ def _osemgrep_format(
 
             engine_label = "PRO" if is_pro else "OSS"
 
+            show_dataflow_traces = extra["dataflow_traces"]
```

Comment:
please use the cli_output_extra field which is typed, so you can access it via cli_output_extra.dataflow_traces so mypy can detect typo.

Reply:
You mean to add it to the atd interface first, then make that change here. Is that correct?

Reply:
I will try to get this PR in.

If you meant to also update the interface and use cli_output_extra, I'll do that as a follow up task.

Reply:
I mean that in python we should avoid to use code that can't be typed by mypy.
If you use dict["some_field"], there is no way mypy can know whether this "some_field" is valid.
This is why we gradualy changed the python code to use the typed interface in semgrep_output_v1.py (generated from semgrep_output_v1.atd).
We migrated lots of the python code to use more types, but there's still a few places where we don't.
But in that case, we have a dataflow_traces field defined in the cli_match_extra so you should use it.